core/op_state.rs

```diff
@@ -0,0 +1,33 @@
+use crate::gotham_state::GothamState;
+use std::ops::Deref;
+use std::ops::DerefMut;
+
+pub struct OpState {
+  pub resource_table: crate::ResourceTable,
+  pub get_error_class_fn: crate::runtime::GetErrorClassFn,
```

Comment:
It's back 🤯  
Why can't we liberate ourselves from the shackles that are C idioms...